app/scripts/controllers/permissions/background-api.js

```diff
@@ -1,38 +1,90 @@
 import nanoid from 'nanoid';
+import { MethodNames } from '@metamask/permission-controller';
+import {
+  Caip25CaveatType,
+  Caip25EndowmentPermissionName,
+} from '../../lib/multichain-api/caip25permissions';
+import {
+  getEthAccounts,
+  setEthAccounts,
+} from '../../lib/multichain-api/adapters/caip-permission-adapter-eth-accounts';
+import { setPermittedEthChainIds } from '../../lib/multichain-api/adapters/caip-permission-adapter-permittedChains';
 import {
   CaveatTypes,
   RestrictedMethods,
 } from '../../../../shared/constants/permissions';
-import { CaveatFactories } from './specifications';
+import { PermissionNames } from './specifications';
 
-export function getPermissionBackgroundApiMethods(permissionController) {
+export function getPermissionBackgroundApiMethods({
+  permissionController,
+  approvalController,
+  networkController,
+}) {
+  // To add more than one account when already connected to the dapp
   const addMoreAccounts = (origin, accountOrAccounts) => {
     const accounts = Array.isArray(accountOrAccounts)
       ? accountOrAccounts
       : [accountOrAccounts];
-    const caveat = CaveatFactories.restrictReturnedAccounts(accounts);
-
-    permissionController.grantPermissionsIncremental({
```

Comment:
I suppose we will want, at some point, to use `grantPermissionsIncremental` again for this kindof permission mutation? Did you attempt to make this work or does the method require modifications to work with CAIP-25 caveats?

Reply:
I didn't try, but I don't think it'll work nicely because you'll need to read the permission in order to know at least one already permitted chainId/scope that you assign the new accounts too

The call would have to look like

```
permissionController.grantPermissionsIncremental({
      subject: { origin },
      approvedPermissions: {
        [Caip25EndowmentPermissionName]: { 
          caveats: [
             { 
                type: Caip25CaveatType,
                value: {
                  requiredScopes: {},
                  optionalScopes: {
                    [existingPermittedScopeThatYouNeedToRead]: {
                      methods: [],
                      notifications: [],
                      accounts: accounts.map((account) => `${existingPermittedScopeThatYouNeedToRead}:account`)
                    }
                  },
                  isMultichainOrigin: false
                }
             
         ]},
      },
})
```

Possibly there's something I'm not considering here. We maybe able to use the [Permission factory and the requestData field](https://github.com/MetaMask/core/blob/8eb7cde27d5ec739d75e9a1654f627e17055d1d7/packages/permission-controller/src/PermissionController.ts#L1779) to build this object for us, but we'd still need to know at least one of the existing permitted chains to pass in.

Perhaps we make this a separate ticket to investigate

Reply:
Agreed lets leave this to a separate investigation!